compose/project.py

```diff
@@ -54,7 +54,7 @@ def visit(n):
     return sorted_services
 
 
-class Project(object):
+class Project(ParallelContainerManager):
```

Comment:
>  You want to avoid inheritance but not giving me any specific reasons why/what problems you can see. 

I see inheritance as the problem.  There is already quite a bit written on the topic ("inheritance vs composition" or "why inheritance is wrong/bad/evil"). You can always represent the same logic using composition (instead of inheritance), and in my experience the code is always more maintainable and easier to read when composition is used.

I don't think this feature necessitates the additional of inheritance (since it's only really sharing a single param that can easily be passed in instead of the service names). It really doesn't even need to be a class, since it's just a single method with no state. It could be just a function.

> Mixins are a common pattern and a good first step to pull this functionality out as we explore further cases of parallelisification

I think this is where we disagree. I see mixins are an anti-pattern, and one that can (and should) be avoided. If this were just a function, you could use it anywhere, without depending on the class having a `containers()` method.  If we want to refactor Project later and move things around, we won't need to modify this container manager, because it doesn't depend on `Project` at all.

> `stop` and `kill` have the same logic for creating the container list, so I'd rather not duplicate that logic

It's not duplicated! you already have a function which handles the logic (`self.containers()`) you're just called a function twice, not duplicating any logic. If you wanted to go a step further and remove the existing duplication, you could make a single method on `Project` that takes the "signal" as a param.  This would removes even more duplication, and still avoid the branch logic and inheritance.